src/FSharp.AWS.DynamoDB/Expression/ConditionalExpr.fs

```diff
@@ -455,54 +483,96 @@ let writeConditionExpression (writer : AttributeWriter) (cond : ConditionalExpre
-        | Contains (attr, op) -> ! "( contains ( " ; !(writer.WriteAttibute attr) ; !", " ; writeOp op ; !" ))"
-        | Attribute_Exists attr -> ! "( attribute_exists ( " ; !(writer.WriteAttibute attr) ; ! "))"
-        | Attribute_Not_Exists attr -> ! "( attribute_not_exists ( " ; !(writer.WriteAttibute attr) ; ! "))"
-        | In(op,ops) -> ! "(" ; writeOp op ; ! " IN (" ; writeOps ops ; ! "))"
+        | False
+        | True -> invalidOp "internal error: invalid query representation"
+        | Not q ->
+            ! "( NOT "
+            writeQuery q
+            ! " )"
+        | And(l, r) ->
+            ! "( "
+            writeQuery l
+            ! " AND "
+            writeQuery r
+            ! " )"
+        | Or(l, r) ->
+            ! "( "
+            writeQuery l
+            ! " OR "
+            writeQuery r
+            ! " )"
+        | Compare(cmp, l, r) ->
+            ! "( "
+            writeOp l
+            writeCmp cmp
+            writeOp r
+            ! " )"
+        | Between(v, l, u) ->
+            ! "( "
+            writeOp v
+            ! " BETWEEN "
+            writeOp l
+            ! " AND "
+            writeOp u
+            ! " )"
+        | BeginsWith(attr, op) ->
+            ! "( begins_with ( "
+            !(writer.WriteAttibute attr)
+            ! ", "
+            writeOp op
+            ! " ))"
+        | Contains(attr, op) ->
+            ! "( contains ( "
+            !(writer.WriteAttibute attr)
+            ! ", "
+            writeOp op
+            ! " ))"
+        | Attribute_Exists attr ->
+            ! "( attribute_exists ( "
+            !(writer.WriteAttibute attr)
+            ! "))"
+        | Attribute_Not_Exists attr ->
+            ! "( attribute_not_exists ( "
+            !(writer.WriteAttibute attr)
+            ! "))"
+        | In(op, ops) ->
+            ! "("
+            writeOp op
+            ! " IN ("
+            writeOps ops
+            ! "))"
 
     writeQuery cond.QueryExpr
     sb.ToString()
 
 /// Generates a conditional that verifies whether a given item exists
-let mkItemExistsCondition (schema : TableKeySchema) =
-    {
-        QueryExpr = AttributeId.FromKeySchema schema |> Attribute_Exists
-        KeyCondition = Some schema
-        NParams = 0
-    }
+let mkItemExistsCondition (schema: TableKeySchema) =
+    { QueryExpr = AttributeId.FromKeySchema schema |> Attribute_Exists
+      KeyCondition = Some schema
+      NParams = 0 }
 
 /// Generates a conditional that verifies whether a given item does not exist
-let mkItemNotExistsCondition (schema : TableKeySchema) =
-    {
-        QueryExpr = AttributeId.FromKeySchema schema |> Attribute_Not_Exists
-        KeyCondition = Some schema
-        NParams = 0
-    }
+let mkItemNotExistsCondition (schema: TableKeySchema) =
+    { QueryExpr = AttributeId.FromKeySchema schema |> Attribute_Not_Exists
+      KeyCondition = Some schema
+      NParams = 0 }
```

Comment:
and I prefer this
but its not consisteny with lack of indenting on space when there are parens...

Reply:
I prefer this style also but I think TaskSeq config specifies Stroustrup style (ie

Reply:
strong agree on not consuming vertical space with trailing ones per stroustrup style
it's not a 2 minute job
I'm up for debating these things and doing it overall but step one is to decide whether to bother, and if so, whether to bother now

Reply:
the best/worst thing is that if it works out and I dont detest it I might ahve no choice but to apply it to github/jet!

Reply:
will delete this comment as review should only ahve things were wemave minor/major diagreements